File: solutions/2. Add Two Numbers.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
​
​
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2){
    struct ListNode *node1 = l1, *node2 = l2, *cur;
    struct ListNode *head = NULL;
    
    int val, flag = 0;
    while(node1 != NULL || node2 != NULL || flag != 0){
        val = 0;
        if(node1 != NULL){
            val += node1->val;
            node1 = node1->next;
        }
        if(node2 != NULL){
            val += node2->val;
            node2 = node2->next;
        }
        val += flag;
        
        flag = val/10;
        val %= 10;
        
        struct ListNode *newNode = (struct ListNode *)malloc(sizeof(struct ListNode));
        newNode->val = val;
        newNode->next = NULL;
        
        if(head == NULL)
            head = newNode;
        else
            cur->next = newNode;
        
        cur = newNode;
    }
    return head;
}

File: solutions/2. Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        result = ListNode(val=0, next=None)
        last = result
        flag = 0
        
        while l1 or l2 or flag:
            val1 = (l1.val if l1 else 0)
            val2 = (l2.val if l2 else 0)
            flag, remainder = divmod(val1+val2+flag, 10)
            
            last.next = ListNode(remainder)
            last = last.next
            
            l1 = (l1.next if l1 else None)
            l2 = (l2.next if l2 else None)
            
        return result.next
